From 0db6fc4824892bc01411cda42ef717de74ec5a09 Mon Sep 17 00:00:00 2001
From: Justin Carpentier <justin.carpentier@inria.fr>
Date: Wed, 10 Jun 2020 10:00:27 +0200
Subject: [PATCH] cmake: remove shared lib

---
 octomap/src/CMakeLists.txt | 35 ++++++++++++++---------------------
 1 file changed, 14 insertions(+), 21 deletions(-)

diff --git a/octomap/src/CMakeLists.txt b/octomap/src/CMakeLists.txt
index 45b384f..472dc44 100644
--- a/octomap/src/CMakeLists.txt
+++ b/octomap/src/CMakeLists.txt
@@ -11,63 +11,56 @@ SET (octomap_SRCS
   )
 
 # dynamic and static libs, see CMake FAQ:
-ADD_LIBRARY( octomap SHARED ${octomap_SRCS})
-set_target_properties( octomap PROPERTIES
-  VERSION ${OCTOMAP_VERSION}
-  SOVERSION ${OCTOMAP_SOVERSION}
-)
 ADD_LIBRARY( octomap-static STATIC ${octomap_SRCS})
 SET_TARGET_PROPERTIES(octomap-static PROPERTIES OUTPUT_NAME "octomap") 
 add_dependencies(octomap-static octomath-static)
 
-TARGET_LINK_LIBRARIES(octomap octomath)
-
 if(NOT EXISTS "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/cmake/octomap")
   file(MAKE_DIRECTORY "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/cmake/octomap")
 endif()
 
-export(TARGETS octomap octomap-static
+export(TARGETS octomap-static
   APPEND FILE "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/cmake/octomap/octomap-targets.cmake")
 
 ADD_SUBDIRECTORY( testing )
 
 ADD_EXECUTABLE(graph2tree graph2tree.cpp)
-TARGET_LINK_LIBRARIES(graph2tree octomap)
+TARGET_LINK_LIBRARIES(graph2tree octomap-static)
 
 ADD_EXECUTABLE(log2graph log2graph.cpp)
-TARGET_LINK_LIBRARIES(log2graph octomap)
+TARGET_LINK_LIBRARIES(log2graph octomap-static)
 
 ADD_EXECUTABLE(binvox2bt binvox2bt.cpp)
-TARGET_LINK_LIBRARIES(binvox2bt octomap)
+TARGET_LINK_LIBRARIES(binvox2bt octomap-static)
 
 ADD_EXECUTABLE(bt2vrml bt2vrml.cpp)
-TARGET_LINK_LIBRARIES(bt2vrml octomap)
+TARGET_LINK_LIBRARIES(bt2vrml octomap-static)
 
 ADD_EXECUTABLE(edit_octree edit_octree.cpp)
-TARGET_LINK_LIBRARIES(edit_octree octomap)
+TARGET_LINK_LIBRARIES(edit_octree octomap-static)
 
 ADD_EXECUTABLE(convert_octree convert_octree.cpp)
-TARGET_LINK_LIBRARIES(convert_octree octomap)
+TARGET_LINK_LIBRARIES(convert_octree octomap-static)
 
 ADD_EXECUTABLE(eval_octree_accuracy eval_octree_accuracy.cpp)
-TARGET_LINK_LIBRARIES(eval_octree_accuracy octomap)
+TARGET_LINK_LIBRARIES(eval_octree_accuracy octomap-static)
 
 ADD_EXECUTABLE(compare_octrees compare_octrees.cpp)
-TARGET_LINK_LIBRARIES(compare_octrees octomap)
+TARGET_LINK_LIBRARIES(compare_octrees octomap-static)
 
 ADD_EXECUTABLE(simple_example simple_example.cpp)
-TARGET_LINK_LIBRARIES(simple_example octomap)
+TARGET_LINK_LIBRARIES(simple_example octomap-static)
 
 ADD_EXECUTABLE(normals_example normals_example.cpp)
-TARGET_LINK_LIBRARIES(normals_example octomap)
+TARGET_LINK_LIBRARIES(normals_example octomap-static)
 
 ADD_EXECUTABLE(intersection_example intersection_example.cpp)
-TARGET_LINK_LIBRARIES(intersection_example octomap)
+TARGET_LINK_LIBRARIES(intersection_example octomap-static)
 
 ADD_EXECUTABLE(octree2pointcloud octree2pointcloud.cpp)
-TARGET_LINK_LIBRARIES(octree2pointcloud octomap)
+TARGET_LINK_LIBRARIES(octree2pointcloud octomap-static)
 
-install(TARGETS octomap octomap-static
+install(TARGETS octomap-static
   EXPORT octomap-targets
   INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
   ${INSTALL_TARGETS_DEFAULT_ARGS}
-- 
2.21.1 (Apple Git-122.3)

